aten/src/ATen/mps/MPSAllocator.mm

```diff
@@ -52,6 +57,21 @@
                   std::make_unique<BufferPool>(m_device, UsageFlags::SMALL | UsageFlags::SHARED | UsageFlags::SCALAR));
 }
 
+BufferPool& MPSHeapAllocatorImpl::get_pool(size_t requested_size, size_t aligned_size, uint32_t usage) {
+  BufferPool::Kind poolKind;
+
+  if (usage & UsageFlags::SCALAR) {
+    poolKind = BufferPool::Kind::SCALAR;
+  } else if (requested_size <= kMaxScalarAlloc && m_device.hasUnifiedMemory) {
```

Comment:
Is this check needed? It seems already covered by next if check

Reply:
How? These are two separate checks:
if (requested_size <= kMaxScalarAlloc ...
and
if (aligned_size <= kMaxSmallAlloc)